README.md

```diff
@@ -123,6 +123,26 @@ let x = 0.0
 end
 ```
 
+Note that the let block is used here to create a scope to allow assignment of 
+`x` within `@byrow!`.
+
+`byrow!` also supports special syntax for allocating new columns to make
+`byrow!` more useful for data transformations. The syntax `@newcol
+x::Array{Int}` allocates a new column `:x` with an `Array` container with eltype
+`Int`. Note that the returned AbstractDataFrame includes these new columns, but
```

Comment:
Thx.

I'm still pondering whether to modify the original. It's easy to change (I've implemented it both ways). The bang is there mainly because the user can overwrite columns. My main reason is that I suspect that future type-stable AbstractDataFrames won't allow `setindex!`; the columns will be fixed. I think this is true of CompositeDataFrames and TypedTables. So, I thought that it would be good to avoid `setindex!` and get used to using the returned value.